runtime/bcutil/cfreader.c

```diff
@@ -765,6 +767,40 @@ readAttributes(J9CfrClassFile * classfile, J9CfrAttribute *** pAttributes, U_32
 			memcpy (stackMap->entries, index, stackMap->mapLength);
 			index += stackMap->mapLength;
 			
+			break;
+		case CFR_ATTRIBUTE_Record:
+			/* JVMS 4.7.30: There may be at most one Record attribute in the attributes table of a ClassFile structure. */
+			if (recordAttributeRead){
+				errorCode = J9NLS_CFR_ERR_MULTIPLE_RECORD_ATTRIBUTES__ID;
+				offset = address;
+				goto _errorFound;
+			}
+			recordAttributeRead = TRUE;
+
+			if (!ALLOC(record, J9CfrAttributeRecord)) {
+				return -2;
+			}
+			attrib = (J9CfrAttribute*)record;
+
+			CHECK_EOF(2);
+			NEXT_U16(record->numberOfRecordComponents, index);
+
+			if (!ALLOC_ARRAY(record->recordComponents, record->numberOfRecordComponents, J9CfrRecordComponent)) {
+				return -2;
+			}
+			for (j = 0; j < record->numberOfRecordComponents; j++) {
+				CHECK_EOF(6);
+				J9CfrRecordComponent* recordComponent = &(record->recordComponents[j]);
+				NEXT_U16(recordComponent->nameIndex, index);
+				NEXT_U16(recordComponent->descriptorIndex, index);
+				NEXT_U16(recordComponent->attributesCount, index);
+				if (!ALLOC_ARRAY(recordComponent->attributes, recordComponent->attributesCount, J9CfrAttribute *)) {
+					return -2;
+				}
+				if ((result = readAttributes(classfile, &(recordComponent->attributes), recordComponent->attributesCount, data, dataEnd, segment, segmentEnd, &index, &freePointer, flags, NULL)) != 0) {
```

Comment:
```suggestion
				result = readAttributes(classfile, &(recordComponent->attributes), recordComponent->attributesCount, data, dataEnd, segment, segmentEnd, &index, &freePointer, flags, NULL);
				if (result != 0) {
```

I know the rest of the code does the assignment in the if but better to do it separately - it aids the readability